subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dependencies/ProjectDependencyInternal.java

```diff
@@ -27,5 +27,8 @@ public interface ProjectDependencyInternal extends ProjectDependency {
      */
     Path getIdentityPath();
 
-    Configuration findProjectConfiguration();
+    /**
+     * Returns the component identifier for the target project.
+     */
+    ProjectComponentIdentifier getProjectComponentIdentifier();
```

Comment:
🤔 This looks a little out of place. I don't think any of the other Dependency types have a component identifier in them. 

Dependency is the input to the resolution process and ComponentIdentifiers are closer to the result of the process.  Could we get by by using the existing `getDependencyProject`?

e.g., `DefaultProjectComponentSelector.newSelector(projectDependency.getDependencyProject());`

Reply:
One of the major motivations behind this deprecation was to remove usages of `getDependencyProject`. 

IMO, I don't think a project dependency should have a `getDependencyProject` method on it at all. Especially in an "Isolated Projects" world. I should not need an instance of a project to create a project dependency. 

A project dependency IMO should just be a wrapper around some unique identifier for a project. According to `ProjectStateRegistry`, projects are uniquely identified either by a `ProjectComponentIdentifier` or a `Path` representing the "identity path" (the build path + the project path within that build).

To me, this is just exposing a unique identifier for the project being referenced by the dependency. 

If I could choose here, I would prefer to use the `getIdentityPath` already exposed from the project dependency, but due to the nature of `ProjectComponentSelector`, we also need to expose the "raw" build path and project path. 

Since these changes are not necessary for the deprecation, I've reverted this, but I think we should be moving _someway_ in this direction. I think we should avoid using `getDependencyProject` and move towards deprecating the method.